interface/reports/inventory_list.php

```diff
@@ -26,46 +26,662 @@
+        "AND s.pid != 0";
+    $binds = array($drug_id, $form_days);
+    if ($warehouse !== '') {
+        $query .= " AND di.warehouse_id = ?";
+        $binds[] = $warehouse;
+    }
+    $srow = sqlQuery($query, $binds);
+    $sales = 0 + $srow['sale_quantity'];
+
+    $query = "SELECT SUM(on_hand) AS on_hand " .
+      "FROM drug_inventory AS di WHERE " .
+      "di.drug_id = ? AND " .
+      "(di.expiration IS NULL OR di.expiration > NOW()) AND " .
+      "di.destroy_date IS NULL";
+    $binds = array($drug_id);
+    if ($warehouse !== '') {
+        $query .= " AND di.warehouse_id = ?";
+        $binds[] = $warehouse;
+    }
+    $ohrow = sqlQuery($query, $binds);
+    $onhand = intval($ohrow['on_hand']);
+
+    if (empty($GLOBALS['gbl_min_max_months'])) {
+        if ($onhand <= $min) {
+            return true;
+        }
+    } else {
+        if ($sales != 0) {
+            $stock_months = sprintf('%0.1f', $onhand * ($form_days / 30.41) / $sales);
+            if ($stock_months <= $min) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+// Generate the list of warehouse IDs that the current user is allowed to access.
+// This is used to build SQL for $uwcond.
+//
+function genUserWarehouses($userid = 0)
+{
+    $list = '';
+    $res = sqlStatement(
+        "SELECT DISTINCT option_id, option_value FROM list_options WHERE " .
+        "list_id = 'warehouse' AND activity = 1"
+    );
+    while ($row = sqlFetchArray($res)) {
+        if (isWarehouseAllowed($row['option_value'], $row['option_id'], $userid)) {
+            if ($list != '') {
+                $list .= ', ';
+            }
+            $list .= "'" . $row['option_id'] . "'";
+        }
+    }
+    return $list;
+}
+
+// This counts the number of days that have a starting zero inventory for a given product
+// since a given start date with given restrictions for warehouse or facility.
+// End date is assumed to be the current date.
+//
+// function zeroDays($product_id, $begdate, $warehouse_id = '~', $facility_id = 0) {
+function zeroDays($product_id, $begdate, $extracond, $min_sale = 1)
+{
+    $today = date('Y-m-d');
+
+    $prodcond = '';
+    if ($product_id) {
+        $prodcond = "AND di.drug_id = '" . add_escape_custom($product_id) . "'";
+    }
+
+    // This will be an array where key is date and value is quantity.
+    // For each date key the value represents net quantity changes for that day.
+    $qtys = array();
+
+    // Force it to have entries for the begin and end dates.
+    $qtys[$today] = 0;
+    $qtys[$begdate] = 0;
+
+    // Get sums of current inventory quantities.
+    $query = "SELECT SUM(di.on_hand) AS on_hand " .
+        "FROM drug_inventory AS di " .
+        "LEFT JOIN list_options AS lo ON lo.list_id = 'warehouse' AND " .
+        "lo.option_id = di.warehouse_id AND lo.activity = 1 " .
+        "WHERE " .
+        "di.destroy_date IS NULL $prodcond $extracond";
```

Comment:
need to ensure $extracond gets escaped in these queries

Reply:
Yes it is.